### PR TITLE
[Updated] Add base price & category to inventory prototype (inventory common)

### PR DESCRIPTION
**Pull request by sboigelot**
feature: #1075

Add category & base price in the inventory prototype (inventory common)

image

### CODE REVIEW

File: Assets/Scripts/Models/Inventory.cs [hunk 1]
```diff
@@ -23,6 +21,7 @@ public class Inventory : IXmlSerializable, ISelectable, IContextActionProvider
     public string objectType = "Steel Plate";
     public int maxStackSize = 50;
     public float basePrice = 1f;
+    public string category = "inv_cat_none";
```
Comment (Grenkin1988) [inline Assets/Scripts/Models/Inventory.cs]: Do you need to edit this field from Unity editor?
If not - make it property instead.
*Best practice - no public fields*

File: Assets/StreamingAssets/Data/Inventory.xml [hunk 2]
```diff
@@ -2,20 +2,32 @@
 <Inventories>
   <Inventory objectType="Raw Iron">
```
Comment (Grenkin1988) [inline Assets/StreamingAssets/Data/Inventory.xml]: ```<Inventory objectType="Raw Iron" maxStackSize="50" basePrice="0.8" category="inv_cat_raw" />```
This looks more ergonomic for me

File: Assets/Scripts/Models/Inventory.cs (hunk 1 above)
Comment (sboigelot) [inline Assets/Scripts/Models/Inventory.cs]: I just follow the document, everything was public field, I added one public field.
But I can change it :) I will when I have access to my computer (and time)

File: Assets/StreamingAssets/Data/Inventory.xml (hunk 2 above)
Comment (sboigelot) [inline Assets/StreamingAssets/Data/Inventory.xml]: Once again, I followed the already setup format of the file. And this isn't relevant as we will probably go away from XML. This one doesn't matter for me. 

Feel free to insist if you really want me to change it. I just need the feature for other PR personally and I know the file format will change soon.

Comment (sboigelot) [inline Assets/StreamingAssets/Data/Inventory.xml]: I'm refactoring this to
```
<Inventories>
  <Inventory objectType="Raw Iron" maxStackSize="50" basePrice="0.8" category="inv_cat_raw"/>
  <Inventory objectType="Ice" maxStackSize="10" basePrice="1.2" category="inv_cat_raw"/>
  <Inventory objectType="Uranium" maxStackSize="20" basePrice="3" category="inv_cat_raw"/>
  <Inventory objectType="Steel Plate" maxStackSize="50" basePrice="1.5" category="inv_cat_refined"/>
  <Inventory objectType="Power Cell" maxStackSize="5" basePrice="5" category="inv_cat_crafted"/>
  <Inventory objectType="Copper Wire" maxStackSize="50" basePrice="2.7" category="inv_cat_crafted"/>
</Inventories>
```

**Comment by vogonistic** [comment]
Merged!

**Comment by Grenkin1988** [comment]
@sboigelot Lots of StyleCop errors related to properties.

**Comment by sboigelot** [comment]
@Grenkin1988 I run my local style cop, I don't find any (in master now). Do you happen to remember what it was saying?

**Comment by Grenkin1988** [comment]
@sboigelot properties after construcotr. And UpperCase for them

**Comment by sboigelot** [comment]
@Grenkin1988 hem oki, in the Inventory.cs, I only changed those because of your comment:

> Do you need to edit this field from Unity editor?
> If not - make it property instead.
> Best practice - no public fields

I will make a PR to fix this.

**Comment by Grenkin1988** [comment]
@sboigelot I've never said that they should be lowerCase :-)

**Comment by sboigelot** [comment]
@Grenkin1988 agree, but it's a dangerous refactoring and it was out of the scope of this PR.
Done in #1157, if you can test it would be great, there is a LOT of changes.

**Pull request merged**